DDD.Functions/EloVotingSubmitPair.cs

```diff
@@ -80,6 +80,23 @@ EloVotingConfig eloVoting
                     return new StatusCodeResult((int)HttpStatusCode.BadRequest);
                 }
             }
+            
+            // only bother doing this check if the configuration value is set to greater than 0, otherwise it's disabled
+            if (eloVoting.EloMinimumThinkingTimeSeconds > 0)
+            {
+                var currentTime = keyDates.Now.ToUnixTimeSeconds();
+                var thinkingTime = currentTime - winnerInUnixTimeSeconds;
+
+                // this introduces a "minimum" time that the user has to wait between submitting a vote, we are effectively forcing
+                // the user to slow down and read the entries - as the payload includes the unix time that the pair was generated we 
```

Comment:
I don't think this is foolproof in combination with the front-end. While the user is comparing one voting pair, we fetch the next behind-the-scenes so that we can present it without any spinners or anything like that. That effectively means that the minimum reading time for that second pair is being used up while you're reading a different pair.

- user visits the page
- presented with pair A, fetches pair B behind the scenes (both pair A and B start their "minimum reading time")
- user spends minimum reading time, then votes on pair A
- user is presented with pair B, and fetches pair C behind the scenes
- user can immediately vote for pair B, because it's minimum reading time has elapsed
- must wait until pair C reading time elapsed to vote for it

I don't think that invalidates the whole approach, as it at least cuts it down on users spamming the vote buttons, but might help us tune the number appropriately/

Reply:
It is a slightly small issue Ben and as you said don't provoke the idea (API rate limit)

Reply:
Could help to think of it as "nobody should be able to vote 3 times within X seconds", because that's what it'll ultimately boil down to without any other changes. Still useful, and would prevent some spamming, but it won't ensure that every pair is reviewed for the minimum amount